lib/ordered_set.h

```diff
@@ -17,234 +17,540 @@ limitations under the License.
 #ifndef _LIB_ORDERED_SET_H_
 #define _LIB_ORDERED_SET_H_
 
+#include <boost/container/set.hpp>
+#include <boost/intrusive/list.hpp>
+#include <boost/tuple/tuple.hpp>
```

Comment:
We should always use `std::tuple` rather than `boost::tuple` for better standards compliance.